README.md

```diff
@@ -2,4 +2,161 @@
+however we recommend using:
+
+- Docker >= 18.02.0 (_older versions have problems during containers start up and/or tear down_)
+- Docker Compose >= 1.17.1
+
+## Usage
+
+SBT
+
+    libraryDependencies += "com.feedzai" %% "cosy-test" % "0.0.3"
+
+Maven
+
+    <dependency>
+        <groupId>com.feedzai</groupId>
+        <artifactId>cosy-test_2.12</artifactId>
+        <version>0.0.3</version>
+    </dependency>
+
+## Example - ScalaTest
+
+``` scala
+class IntegrationSpec extends FlatSpec with DockerComposeTestSuite with MustMatchers {
+
+  override def dockerSetup = Some(
+    DockerComposeSetup(
+      "scalatest",
+      Seq(Paths.get("src", "test", "resources", "docker-compose-scalatest.yml")),
+      Paths.get("").toAbsolutePath,
+      Map.empty
+    )
+  )
+
+  behavior of "Scala Test"
+
+  it must "Retrieve all services" in {
+    val expectedServices = Set("container1", "container2", "container3")
+    dockerSetup.foreach { setup =>
+      setup.getServices().toSet mustEqual expectedServices
+    }
+  }
+}
+```
+
+
+## Example - JUnit
+
+``` java
+public class IntegrationSpec {
+
+    private static final DockerComposeJavaSetup dockerSetup;
+
+    static {
+        dockerSetup = new DockerComposeJavaSetup(
+            "junittest",
+            Collections.singletonList(Paths.get("src", "test", "resources", "docker-compose-junit.yml")),
+            Paths.get("").toAbsolutePath(),
+            new HashMap<>()
+        );
+    }
+
+    @ClassRule
+    public static DockerComposeRule dockerComposeRule = new DockerComposeRule(dockerSetup);
+
+    @Rule
+    public TestWatcher testWatcher = new TestWatcher() {
+        @Override
+        protected void failed(Throwable e, Description description) {
+            dockerComposeRule.setTestFailed(true);
+        }
+    };
+
+    @Test
+    public void fetchServices() {
+       Assert.assertThat(
+           dockerSetup.getServices(),
+           containsInAnyOrder("container1", "container2", "container3")
+       );
+    }
+}
+```
+
+
+## Example - Gatling
+
+``` scala
+class IntegrationSpec extends DockerComposeSimulation {
+
+  override def dockerSetup = Some(
+    DockerComposeSetup(
+      "gatling",
+      Seq(Paths.get("src", "test", "resources", "docker-compose-gatling.yml")),
+      Paths.get("").toAbsolutePath,
+      Map.empty
+    )
+  )
+
+  beforeSimulation()
+
+  private val HttpProtocol: HttpProtocolBuilder = http
+    .baseURL("http://localhost:8086")
+
+  private val populationBuilder = {
+    val getAction = http("Gatling simulation").get("")
+    val s = scenario("Gatling simulation")
+    s.during(10.seconds)(feed(Iterator.empty).exec(getAction))
+      .inject(rampUsers(10).over(1.second))
+      .protocols(HttpProtocol)
+  }
+
+  setUp(populationBuilder)
+}
+```
+
+## Build, Test and Package
+
+After cloning the repository you can simply build and run the tests, by executing the command:
+
+        sbt test
+
+And to generate binaries:
+
+        sbt package
```

Comment:
I am not sure this is useful. I would rather add a comment on versioning (that it’s using dynver and derives the version from the tag).